modules/payment/src/PluginForm/OffsitePaymentForm.php

```diff
@@ -0,0 +1,43 @@
+<?php
+
+namespace Drupal\commerce_payment\PluginForm;
+
+use Drupal\commerce_payment\Plugin\Commerce\PaymentGateway\OffsiteRedirectPaymentGatewayInterface;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\Core\Render\Element;
+
+abstract class OffsitePaymentForm extends PaymentGatewayFormBase {
```

Comment:
@bojanz should we document that all off-site payments _must_ extend this and they're expected to override `buildConfigurationForm` and enter any data to be POST'd to the gateway? Or do handhsake and throw `NeedsRedirectException` in their form build?